storage: `BenchmarkTimeBoundIterate/LoadFactor=1.00/TimeBoundIterator` fails

```
./dev bench --short --bench-time=1x pkg/ccl/storageccl/engineccl -f 'BenchmarkTimeBoundIterate/LoadFactor=1.00/TimeBoundIterator' --stream-output --ignore-cache
$ bazel test pkg/ccl/storageccl/engineccl:all --nocache_test_results --test_arg -test.run=- --test_arg -test.bench=BenchmarkTimeBoundIterate/LoadFactor=1.00/TimeBoundIterator --test_sharding_strategy=disabled --test_arg -test.cpu --test_arg 1 --test_arg -test.short --test_arg -test.benchtime=1x --crdb_test_off --test_env COCKROACH_TEST_FIXTURES_DIR=/home/jackson_cockroachlabs_com/.cache/crdb-test-fixtures --sandbox_writable_path=/home/jackson_cockroachlabs_com/.cache/crdb-test-fixtures --test_output streamed
INFO: Invocation ID: b82e60e3-5e51-47f6-8a7c-9114c3e47e70
WARNING: Streamed test output requested. All tests will be run locally, without sharding, one at a time
INFO: Analyzed 2 targets (0 packages loaded, 0 targets configured).
I241028 20:48:22.013512 67 ccl/storageccl/engineccl/bench_test.go:113  [-] 1  using test data: /home/jackson_cockroachlabs_com/.cache/crdb-test-fixtures/mvcc_data_v2_vv1000024.2-upgrading-to-1000024.3-step-022_100000_100_10_512
I241028 20:48:22.017718 67 3@pebble/open.go:1021  [s?,pebble] 2  [JOB 1] WAL 000535 stopped reading at offset: (/home/jackson_cockroachlabs_com/.cache/crdb-test-fixtures/mvcc_data_v2_vv1000024.2-upgrading-to-1000024.3-step-022_100000_100_10_512/000535.log: 0); replayed 0 keys in 0 batches
I241028 20:48:22.017860 71 3@pebble/event.go:784  [s?,pebble] 3  [JOB 2] flushing 1 memtable (0B) to L0
I241028 20:48:22.020777 71 3@pebble/event.go:804  [s?,pebble] 4  [JOB 2] MANIFEST created 000538
I241028 20:48:22.020834 71 3@pebble/event.go:788  [s?,pebble] 5  [JOB 2] flush error: pebble: empty table
I241028 20:48:22.022018 67 3@pebble/event.go:832  [s?,pebble] 6  [JOB 1] WAL created 000539
I241028 20:48:22.028674 69 3@pebble/event.go:836  [s?,pebble] 7  [JOB 2] WAL deleted 000535
I241028 20:48:22.028820 69 3@pebble/event.go:808  [s?,pebble] 8  [JOB 1] MANIFEST deleted 000530
I241028 20:48:22.075055 78 3@pebble/event.go:824  [s?,pebble] 9  [JOB 4] all initial table stats loaded
--- FAIL: BenchmarkTimeBoundIterate/LoadFactor=1.00/TimeBoundIterator
    test_fixtures.go:85: using existing fixture "mvcc_data_v2_vv1000024.2-upgrading-to-1000024.3-step-022_100000_100_10_512" in "/home/jackson_cockroachlabs_com/.cache/crdb-test-fixtures/mvcc_data_v2_vv1000024.2-upgrading-to-1000024.3-step-022_100000_100_10_512"
    bench_test.go:170: expected at least 100000 keys, but got 99999
--- FAIL: BenchmarkTimeBoundIterate/LoadFactor=1.00
--- FAIL: BenchmarkTimeBoundIterate
FAIL
I241028 20:48:22.096579 1 (gostd) testmain.go:113  [-] 1  Test //pkg/ccl/storageccl/engineccl:engineccl_test exited with error code 1
```

The time-bound iterator with the most inclusive bounds is somehow missing a key.

Jira issue: CRDB-43704

--- a/pkg/ccl/storageccl/engineccl/bench_test.go
+++ b/pkg/ccl/storageccl/engineccl/bench_test.go
@@ -94,6 +94,9 @@ func loadTestData(
 				minWallTime = minSStableTimestamps[i/scaled]
 			}
 			timestamp := hlc.Timestamp{WallTime: minWallTime + rand.Int63n(int64(batchTimeSpan))}
+			if timestamp.Less(hlc.MinTimestamp) {
+				timestamp = hlc.MinTimestamp
+			}
 			value := roachpb.MakeValueFromBytes(randutil.RandBytes(rng, valueBytes))
 			value.InitChecksum(key)
 			if _, err := storage.MVCCPut(ctx, batch, key, timestamp, value, storage.MVCCWriteOptions{}); err != nil {

